pymediainfo/__init__.py

```diff
@@ -346,11 +346,44 @@ def parse(cls, filename, library_file=None, cover_data=False,
                 )
             for option_name, option_value in mediainfo_options.items():
                 lib.MediaInfo_Option(handle, option_name, option_value)
-        if lib.MediaInfo_Open(handle, filename) == 0:
-            lib.MediaInfo_Close(handle)
-            lib.MediaInfo_Delete(handle)
-            raise RuntimeError("An error occured while opening {}"
-                    " with libmediainfo".format(filename))
+        try:
+            filename.seek(0, 2)
+            file_size = filename.tell()
+            filename.seek(0)
+        except AttributeError:  # filename is not a file-like object
+            file_size = None
+
+        if file_size is not None:  # We have a file-like object, use the buffer protocol:
+            if 'b' not in getattr(filename, 'mode', 'b'):
+                raise ValueError("file should be opened in binary mode")
+            lib.MediaInfo_Open_Buffer_Init(handle, file_size, 0)
```

Comment:
Should we handle return codes from these calls? I assume if an error occurs, they will return zero similarly to `MediaInfo_Open`?

Reply:
No, it just returns the pos, and in upstream code it's never checked as far as I can read:
```cpp
size_t MediaInfoList_Internal::Open_Buffer_Init (int64u File_Size_, int64u File_Offset_)
{
    MediaInfo_Internal* MI=new MediaInfo_Internal();
    MI->Open_Buffer_Init(File_Size_, File_Offset_);

    CriticalSectionLocker CSL(CS);
    size_t Pos=Info.size();
    Info.push_back(MI);
    return Pos;
}
```
So I bet there's nothing to check here.